cai_causal_graph/causal_graph.py

```diff
@@ -741,6 +741,9 @@ def _set_edge(self, edge: Edge, validate: bool = True):
             edge.destination
         )
 
+        if source in self._edges_by_source[destination].keys():
```

Comment:
Slightly faster to do the following, because you won't need to unpack the keys:

```python
if self._edges_by_source[destination].get(source) is not None:
    raise CausalGraphErrors.ReverseEdgeExistsError()
```